### ISSUE TITLE
Include "Your Water Usage" information in the sensor

### ISSUE THREAD

**Issue by aefo** [sHedC/homeassistant-leakbot#82]
### Checklist

- [X] I have filled out the template to the best of my ability.
- [X] This only contains 1 feature request (if you have multiple feature requests, open one feature request for each feature request).
- [X] This issue is not a duplicate feature request of [previous feature requests](https://github.com/sHedC/homeassistant-leakbot/issues?q=is%3Aissue+label%3A%22Feature+Request%22+).

### Is your feature request related to a problem? Please describe.

Would be great if the sensor could include the "your water usage" (High/Medium/Low) used on previous days.  (As displayed in the app)

### Describe the solution you'd like

Even when I have 0 leak free days, I'd like to be able to check that the water usage is still "Low" (and not jumped up suddenly)

### Describe alternatives you've considered

n/a (checking the app)

### Additional context

n/a

**Comment by sHedC** [comment]
Sorry for being slow have been super busy,  I have started to look at enhancing some of my projects again.

**Comment by stephanschn** [comment]
+1 from me

**Comment by Hoekeren** [comment]
+1 from me aswell

**Comment by sHedC** [comment]
I am not sure how to implement this with HomeAssistant I can use a single indicator, the way the API gives it has the history in the API which calculates in the App to last 7 days, month etc.

But this doesn't really translate into home assistant with history so struggling to think how to work this any ideas?

**Comment by aefo** [comment]
Do you happen to have a example of the response / information returned the API? (is is a numeric value, or an enum of High/Medium/Low) 

As a first thought, giving a single sensor value of the 'latest' value from the history would probably be useful. The "history" would then recorded in HA logbook.

My goal / use case is primarily just to set an alarm /automation on the value not being high.

Thanks

**Comment by sHedC** [comment]
If you check in tests/fixtures there is examples of the API and the water usage one.

https://github.com/sHedC/homeassistant-leakbot/tree/main/tests/fixtures

**Comment by aefo** [comment]
> If you check in tests/fixtures there is examples of the API and the water usage one.
> 
> https://github.com/sHedC/homeassistant-leakbot/tree/main/tests/fixtures

Thanks, the response information is more detailed than I expected. Great to have it broken down by period of the day & numerically. 

Will look to thing of ways to present usefully in HA, but currently just having each value of "latest" exposed in some format would be helpful.

**Comment by aefo** [comment]
Not sure if https://github.com/ldotlopez/ha-historical-sensor is helpful as an approach to be able to write in historical sensor values

**Comment by sHedC** [comment]
> Not sure if https://github.com/ldotlopez/ha-historical-sensor is helpful as an approach to be able to write in historical sensor values

That looks interesting I will have a look

**Comment by sHedC** [comment]
Progress, not showing in the Energy Monitor and also not sure if it will retain history as I add, but a start.

![Image](https://github.com/user-attachments/assets/85d7d331-0da6-428f-95f0-2dbe6d0d890f)

**Comment by sHedC** [comment]
Right gone through this, the measurements in Leakbot are based on detecting usage per 30 min unit, so the maximum count is 48 per 24 hours.

This means it does not detect usage only 30 min units used, could estimate for the energy usage that would be 10l per min at 1 bar so usual house anything from 20-30l per min which means measuring 600 to 900l per min which is not useful for the energy monitor.

I can add history for minutes or hours detected for morning/ afternoon/ evening/ night but won't show in the energy monitor.

**Comment by sHedC** [comment]
@aefo @Hoekeren @stephanschn 

The Water Usage is calculated based on each half hour it is detected that water is running per day, so in each day there are 48 units.  The units are based on 6 hour periods, night, morning, afternoon, evening.  They are updated after 2 days, so the latest stats are always 2 days out of date.

This can be shown either as half hour units or litres.  
- Litres - Show in History and in Energy under water (but its not correct)
- Hours - Show in History Only.

Based on this is it even worth showing in the energy, there are better water measure devices out?

Thoughts?

History:

![Image](https://github.com/user-attachments/assets/826c14d1-4c44-42d6-9b62-6d98273d2bca)

Energy (if showing in Litres)

![Image](https://github.com/user-attachments/assets/3dec4647-7b85-4abc-b3c2-b60cea443f77)

**Comment by stephanschn** [comment]
The values delivered by LeakBot seem to be only qualitative. I would prefer to get only the numerical value out of the sensor as devliered by the API. If needed, a helper entity could map this into water or energy consumption.

**Comment by sHedC** [comment]
> The values delivered by LeakBot seem to be only qualitative. I would prefer to get only the numerical value out of the sensor as devliered by the API. If needed, a helper entity could map this into water or energy consumption.

Yes I put an example of Litres to show what it looked like in the energy consumption, if I use Units then it won't show int he Energy console.

**Comment by sHedC** [comment]
Updated to 1.0.0 Beta 0 for testing

**Comment by aefo** [comment]
Thanks for the updates. Looking forward to testing it.

**Comment by stephanschn** [comment]
> Updated to 1.0.0 Beta 0 for testing

Looks good. I'll give it some days to compare the data with the app and real world consumption. Thanks!

**Comment by aefo** [comment]
Updated as well, thanks

**Comment by sHedC** [comment]
All seems to be working for me, and its retaining the previous days in the history, the history available is only the previous 30 days.

All noticing working well?

**Comment by stephanschn** [comment]
looks good to me... great job!
![Image](https://github.com/user-attachments/assets/cbabdd9c-d6d2-4aee-989a-c8c43509accb)

**Comment by sHedC** [comment]
> looks good to me... great job! ![Image](https://github.com/user-attachments/assets/cbabdd9c-d6d2-4aee-989a-c8c43509accb)

Great I will merge it with main, I am working on a high water alert notification as I got one today so want to see how to add that too. #168
